beard/similarity/pairs.py

```diff
@@ -273,3 +274,48 @@ def transform(self, X):
         X2 = X[:, n_features:]
 
         return np.abs(X1 - X2)
+
+
+class JaccardSimilarity(BaseEstimator, TransformerMixin):
+
+    """Similarity as the intersection between 2 sets over the union of them."""
+
+    def fit(self, X, y=None):
+        """(Do nothing).
+
+        Parameters
+        ----------
+        :param X: array-like, shape (n_samples, n_features) Input data.
+        Returns
+        -------
+        :returns: self
+        """
+        return self
+
+    def transform(self, X):
+        """Compute the intersection over the union.
+
+        Get the intersection and the union of the 2 feature sets
+        and compute the Jaccard similarity as intersection over union
+        Here, since we have vectorized representation of the input, we
+        calculate the intersection as the dot product and the union as
+        the count of non_zeros of the addition of the 2 vectors.
+
+        Parameters
+        ----------
+        :param X: array-like, shape (n_samples, n_features)
+            Input data.
+
+        Returns
+        -------
+        :returns: Xt which is an array of similarity values
+        """
+        X = sp.csr_matrix(X)
+        n_samples, n_features_all = X.shape
+        n_features = n_features_all // 2
+        X1 = X[:, :n_features]
+        X2 = X[:, n_features:]
+        numerator = np.asarray(X1.multiply(X2).sum(axis=1)).ravel()
+        denominator = (X1 + X2).getnnz(axis=1)
```

Comment:
Wouldnt it crash if you divide by zero? Please add a test for that to ensure the proper behaviour.